src/log_batch.rs

```diff
@@ -555,6 +555,14 @@ enum BufState {
     /// # Invariants
     /// LOG_BATCH_HEADER_LEN <= buf.len()
     Sealed(usize, usize),
```

Comment:
Let's not add yet another state. Change `Sealed` to contain a third field "original checksum". During `prepare_write`, if the state is `Sealed`, we can use this field to re-sign checksum.

Reply:
`ReSealing` is a special state to handle the recursive `write` problem, remarking that the LogBatch has entered the `rewrite` state. 

With this special state, the log_batch will follow the state transformating path `Write(Open -> Encoded -> Sealed)` => `Rewrite(Sealed -> Resealing -> end with panic || end with successful "sync")`

So, if `rewrite` failed,  the next `return self.write(log_batch, sync)` would `panic` in the next checking of `log_batch.prepare_rewrite()`.

Reply:
It's too obscure to use this as a recursion limit. Using a loop to retry is more explicit and understandable.